spring-ai-core/src/main/java/org/springframework/ai/model/observation/ErrorLoggingObservationHandler.java

```diff
@@ -62,12 +64,13 @@ public ErrorLoggingObservationHandler(Tracer tracer,
 	}
 
 	@Override
-	public boolean supportsContext(Context context) {
-		return (context == null) ? false : this.supportedContextTypes.stream().anyMatch(clz -> clz.isInstance(context));
+	public boolean supportsContext(@Nullable Context context) {
+		return context != null && this.supportedContextTypes.stream().anyMatch(clz -> clz.isInstance(context));
 	}
 
 	@Override
-	public void onError(Context context) {
+	@SuppressWarnings("unused")
+	public void onError(@Nullable Context context) {
```

Comment:
Same thing here, the `Context` is never null as per the Micrometer infrastructure.

Reply:
Same comment as above.